CONTRIBUTING.md

```diff
@@ -1,35 +1,224 @@
+
+## Before submitting an enhancement/feature request
+
+*   Check the [wiki][wiki-url] and [forum][forum-url] for tips — you might discover that the enhancement is already available.
+*   Most importantly, check if you're using the latest version of qBittorrent and if you can get the desired behavior by changing qBittorrent's settings.
+*   Check in the [releases][releases-url] page or on the [forum][forum-url], see if there's already a alpha/beta version with that enhancement.
+*   Perform a cursory search to see if the enhancement has already been suggested. If it has, add a comment to the existing issue instead of opening a new one.
+
+## Steps to ensure a good enhancement/feature suggestion
+
+-   Specify which version of qBittorrent you're using.
+-   Specify the name and version of the OS you're using.
+-   Provide a step-by-step description of the suggested enhancement in as many details as possible.
+-   Describe the current behavior and explain which behavior you expected to see instead and why.
+-   Include screenshots and animated GIFs which help you demonstrate the steps or point out the part of qBittorrent which the suggestion is related to.
+-   If this enhancement exists in other bittorrent clients, list those clients.
 
 # Opening a pull request
 
 ### Must read
-* Read our [**coding guidelines**](https://github.com/qbittorrent/qBittorrent/blob/master/CODING_GUIDELINES.md). There are some scripts to help you: [uncrustify script](https://raw.githubusercontent.com/qbittorrent/qBittorrent/master/uncrustify.cfg), [astyle script](https://gist.github.com/Chocobo1/539cee860d1eef0acfa6), [(related thread)](https://github.com/qbittorrent/qBittorrent/issues/2192).
-* Keep the title **short** and provide a **clear** description about what your pull request does.
-* Provide **screenshots** for UI related changes.
-* Keep your git commit history **clean** and **precise**. Commits like `xxx fixup` should not appear.
-* If your commit fix a reported issue (for example #4134), add the following message to the commit `Closes #4134.`. Example [here](https://github.com/qbittorrent/qBittorrent/commit/a74bac20c4e8de9776bf9bb77fdc7526135d1988).
+*   Read our [**coding guidelines**][coding-guidelines-url]. There are some scripts to help you: [uncrustify script][uncrustify-script-url], [astyle script][astyle-script-url], [(related thread)][coding-guidelines-thread-url].
+*   Keep the title **short** and provide a **clear** description about what your pull request does.
+*   Provide **screenshots** for UI related changes.
+*   Keep your git commit history **clean** and **precise.** Commits like `xxx fixup` should not appear.
+*   If your commit fixes a reported issue (for example #4134), add the following message to the commit `Closes #4134.`. Example [here][commit-message-fix-issue-example-url].
+*   Write good commit messages.
+    *   BAD EXAMPLE: `fixed stuff`
+    *   GOOD EXAMPLE: `fixed <a certain issue> by doing this and that`
```

Comment:
As per [the new rule about commit messages](https://github.com/qbittorrent/qBittorrent/blob/master/CODING_GUIDELINES.md#10-git-commit-message)
you should change "fixed" to "Fix" (imperative mood and capitalize).

Reply:
Thanks, fixed